src/hooks/useFocusWhenVisible.ts

```diff
@@ -0,0 +1,63 @@
+import { RefObject, useEffect } from 'react';
+
+const useFocusWhenVisible = (
+  ref: RefObject<HTMLElement | null>,
+  enabled: boolean = true,
+  maxWaitTime: number = 1000,
+) => {
```

Comment:
Let's go with named arguments over positional ones:


```suggestion
const useFocusWhenVisible = ({ 
  ref: RefObject<HTMLElement | null>,
  enabled: boolean = true,
  maxWaitTime: number = 1000,
}: {
  ref: RefObject<HTMLElement | null>;
  enabled?: boolean;
  maxWaitTime?: number;
}) => {
```

It's easier to read:
```useFocusWhenVisible({ ref, enabled: true })```
compared to:
```useFocusWhenVisible(ref, true)```

Reply:
Done. Thanks!